### PR TITLE
Enable more passing win unit tests

### PR DESCRIPTION
**Pull request by andrewsavage1**
b/330792170

### CODE REVIEW

File: base/threading/platform_thread.h
Comment (andrewsavage1) [inline base/threading/platform_thread.h:1]: @johnxwork I'm looking for thoughts on what's best to do here, as the ordering of thread types is different here than it seemed to be before: https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:base/threading/platform_thread.h;l=124-132?q=platform_thread.h&ss=h%2Flbshell-internal%2Fcobalt_src%2F%2B%2Frefs%2Fheads%2FCOBALT

File: base/threading/platform_thread.h [hunk 1]
```diff
@@ -123,6 +119,10 @@ enum class ThreadType : int {
   // Suitable for low-latency, glitch-resistant audio.
   kRealtimeAudio,
   kMaxValue = kRealtimeAudio,
+  // Default type. The thread priority or quality of service will be set to
+  // platform default. In Chrome, this is suitable for handling user
+  // interactions (input), only display and audio can get a higher priority.
+  kDefault = kSbThreadNoPriority,
```
Comment (sherryzy) [inline base/threading/platform_thread.h:125]: I don't have enough context on the thread priority change, but shall we gate this with a macro, eg: COBALT_PENDING_CLEAN_UP?

Comment (johnxwork) [reply base/threading/platform_thread.h:125]: I think your logic is correct. The Starboard API defined kSbThreadNoPriority as the default priority on each platform, which should be close to Chrome's definition for kDefault.

But I think the simple casting in platform_thread_starboard.cc is very dangerous. Can we keep this file intact but do explicit Chromium->Starboard priority mapping in platform_thread_starboard.cc?

File: base/threading/platform_thread.h
Comment (johnxwork) [reply base/threading/platform_thread.h:1]: Please see my in-file comment. In old base, we defined our priority in platform_thread.h. This is removed now as it should be. It was dangerous and makes upstreaming more difficult. We can do the mapping in the cc file.

File: base/threading/platform_thread.h (hunk 1 above)
Comment (andrewsavage1) [reply base/threading/platform_thread.h]: Added switch statement with mapping for new ordering of priorities

**Comment by andrewsavage1** [comment]
Only failures are flakes

**Review by johnxwork — APPROVED** [review]

**Pull request merged**